Sources/WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -141,19 +141,30 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
 
         switch authorizationStatus {
         case .authorized:
-            DispatchQueue.main.async {
-                self.captureSession.startRunning()
+            DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+                self?.captureSession.startRunning()
+                DispatchQueue.main.async {
+                    self?.isDetecting = true
+                }
             }
-            isDetecting = true
         case .notDetermined:
-            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { _ in
-                DispatchQueue.main.async { [weak self] in
-                    self?.start()
+            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { [weak self] granted in
+                if granted {
+                    DispatchQueue.main.async {
+                        self?.start()
+                    }
+                } else {
+                    DispatchQueue.main.async {
+                        let error = ImageScannerControllerError.authorization
+                        self?.delegate?.captureSessionManager(self!, didFailWithError: error)
```

Comment:
It would probably be safer to `guard let self else { return }` to prevent having to unwrap `self` here, as well as in the `default` case.

Reply:
Your suggestion is indeed a safer approach to avoid force unwrapping self in these closure contexts, ensuring that we gracefully handle the scenario where self might be nil.